uqcsbot/snailrace.py

```diff
@@ -0,0 +1,60 @@
+import discord, asyncio
+from discord import app_commands, ui
+
+from discord.ext import commands
+from uqcsbot.bot import UQCSBot
+
+import uqcsbot.utils.snailrace_utils as snail
+
+
+# Trying out Discord buttons for Snail Race Interactions
+class SnailRaceView(discord.ui.View):
+    def __init__(self, raceState: snail.SnailRaceState):
+        super().__init__(timeout=snail.SNAILRACE_OPEN_TIME)
+        self.raceState = raceState
+    
+    async def on_timeout(self):
+        """
+        Called when the view times out. This will deactivate the buttons and
+        begine the race.
+        """
+        for child in self.children:
+            child.disabled = True
+        await self.raceState.open_interaction.edit_original_response(content=snail.SNAILRACE_ENTRY_CLOSE, view=self)
+        await self.raceState.race_start()
+
+    @ui.button(label="Enter Race", style=discord.ButtonStyle.primary)
+    async def button_callback(self, interaction: discord.Interaction, button: discord.ui.Button):
+        if self.raceState.add_racer(interaction.user):
```

Comment:
General comment:

Keen to see what sort of size races get to on the main server... I would suggest capping at 35 entrants, which is the rough height of my (admittedly not massive) laptop screen's message display capabilities, but that's not worth the effort considering the number of people I expect to participate in snailraces.

Reply:
Thats fair, I'm gonna limit it to 12 for now, in tradition to horse racing (I'm not addicted to punting... you are).

Might amend this later with an optional param to set race sizes in the `/snailrace` command.

Reply:
Massive races have some appeal to them (provided they still fit on most screens). 12 is a fair limit, and adding configurability (via argument or just changing the max via PR) is always good.